core/lib/zksync_core/src/state_keeper/io/common.rs

```diff
@@ -134,12 +133,31 @@ pub(crate) async fn load_pending_batch(
             .expect("`protocol_version` must be set for pending miniblock"),
         pending_miniblock_header.virtual_blocks,
         chain_id,
-    );
+    ))
+}
+
+/// Loads the pending L1 block data from the database.
+pub(crate) async fn load_pending_batch(
+    storage: &mut StorageProcessor<'_>,
+    current_l1_batch_number: L1BatchNumber,
+    fee_account: Address,
+    validation_computational_gas_limit: u32,
+    chain_id: L2ChainId,
+) -> Option<PendingBatchData> {
+    let (system_env, l1_batch_env) = load_l1_batch_params(
+        storage,
+        current_l1_batch_number,
+        fee_account,
+        validation_computational_gas_limit,
+        chain_id,
+    )
+    .await?;
 
     let pending_miniblocks = storage
         .transactions_dal()
         .get_miniblocks_to_reexecute()
-        .await;
+        .await
+        .unwrap();
```

Comment:
Sneaky unwrap be sneaky.

Reply:
Oh right. I noticed the `?` before and didn't understand that it's `Try` on `Option`, not on `Result` 🤦 
Fine to keep it (for now).